File: Transport front Final (2)/src/app/mode/Commande.ts
import {Chauffer} from "./Chauffer";
import {Client} from "./Client";

export class Commande {
  id: number;
  depart: string;
  arrive: string;

  carType: CarType;
  numberPersons: number;
  dateDepart: Date;
  price: number;
  distance: number;
  stat: Stat;

  chauffeur: Chauffer;

  client: Client;
}

export enum Stat {
  EN_COURS,EN_ATTENTE,COMPLETE,ANNULER
}

export enum CarType {
  VEHICULE,CAMION,CAMION_POIDS_LOURDS

}

File: transport-main F (1)/transport-main F/transport-main/src/main/java/com/transport/enteties/Commande.java
package com.transport.enteties;

import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import javax.persistence.*;
import java.util.Date;

@Entity
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
public class Commande {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    private String depart;
    private String arrive;
    private CarType carType;
    private int numberPersons;
    private Date dateDepart;
    private float price;
    private  float distance;
    private Stat stat;
    @ManyToOne
    private Chauffeur chauffeur;
    @ManyToOne
    private Client client;

}
